powershell/public/Send-MtTeamsChannelMessage.ps1

```diff
@@ -0,0 +1,153 @@
+    must be granted to the application in the Microsoft Entra portal.
+
+.EXAMPLE
+    Send-MtTeamsChannelMessage -MaesterResults $MaesterResults -TeamId <guid> -ChannelId <guid> -Subject 'Maester Results' -TestResultsUri "https://github.com/contoso/maester/runs/123456789"
+
+    Sends an Adaptive Card in a Teams Channel with the summary of the Maester test results to the specified channel along with the link to the detailed test results.
+#>
+
+Function Send-MtTeamsChannelMessage {
+    [CmdletBinding()]
+    param(
+        # The Maester test results returned from `Invoke-Pester -PassThru | ConvertTo-MtMaesterResult`
+        [Parameter(Mandatory = $true, Position = 0)]
+        [psobject] $MaesterResults,
+
+        # The team id of where the message should be posted. e.g. 5ba5cbbb-675e-4a8d-a382-a52960a18b4d
+        [Parameter(Mandatory = $true, Position = 1)]
+        [string[]] $TeamId,
+
+        # The channel id of where the message should be posted. e.g. 19:4a95f7d8db4c4e7fae857bcebe0623e6@thread.tacv2
+        [Parameter(Mandatory = $true, Position = 2)]
+        [string[]] $ChannelId,
+
+        # The subject of the card. Defaults to 'Maester Test Results'.
+        [string] $Subject,
+
+        # Uri to the detailed test results page.
+        [string] $TestResultsUri
+
+    )
+
+    <#
+        # Developer guide for editing the Adaptive Card.
+        - Authoring of the Adaptive Card is done using the Adaptive Card Designer.
+        - Card Payload can be found in /powershell/assets/AdaptiveCard.json
+        - Open https://adaptivecards.io/designer/, insert the payload and example data ($adaptiveCardData | ConvertTo-Json -depth 5)
+        - Make changes as needed
+        - Copy the payload and paste it into the /powershell/assets/AdaptiveCardPayloadTemplate.json file in the assets folder
+
+        To do:
+        - Add a switch to send the card to a user instead of a channel
+    #>
+    if (!(Test-MtContext -SendChannelMessage)) { return }
+
+    if (!$Subject) { $Subject = "Maester Test Results" }
+
+    $adaptiveCardTemplateFilePath = Join-Path -Path $PSScriptRoot -ChildPath '../assets/AdaptiveCardPayloadTemplate.json'
+    $adaptiveCardTemplate = Get-Content -Path $adaptiveCardTemplateFilePath -Raw
+
+    $CurrentVersion = $MaesterResults.CurrentVersion
+    $LatestVersion = $MaesterResults.LatestVersion
+    $ModuleVersion =
+        if ($currentVersion -ne $latestVersion) {
+            "$currentVersion → Latest version: $latestVersion"
+        } else {
+            "$currentVersion"
+        }
+
+    $notRunCount = $MaesterResults.SkippedCount
+    if ([string]::IsNullOrEmpty($MaesterResults.SkippedCount)) { $notRunCount = "-" }
+
+    $adaptiveCardData = @{
+            title       = $Subject
+            description = "Results for Maester Test run of $($MaesterResults.ExecutedAt)"
+            run         = @{
+                tenantName    = $MaesterResults.TenantName
```

Comment:
Can we keep the variable name case uniform? Some start with lowercase (e.g. moduleVersion) and others start with upper case (e.g. NotRunCount).